src/Downloader.ts

```diff
@@ -232,6 +232,14 @@ class Downloader {
       await this.initLocalServices();
   }
 
+  public getBaseUrl(articleId: string): string{
+    return `${this.baseUrl}${encodeURIComponent(articleId)}`
+  }
+
+  public getBaseUrlForMainPage(articleId: string): string{
+    return `${this.baseUrlForMainPage}${encodeURIComponent(articleId)}`
+  }
```

Comment:
I understand the logic behind this change and this is not wrong. The problem I see is that the concept of "base url" is wrong. Let me try to explain the reason.

By letting the Downloader deciding how look the URLs we create a situation where it has to decide about the proper approach beetween what is available (Downloader), how the URL looks like (Mediawiki) and how they are treated (the Renderers). It fails to do that properly because most of the constraints are outside its control. This create the kind of incoherencies where a piece of code get an URL... but ultimatively it does not know what kind of URL it is (which is then hard/impossible to deal properly with).

Long story short, I believe we should keep this like it is... even wrong... we will have to change it soon anyway and this will imply its removal.